src/components/controls/TextArea/TextArea.tsx

```diff
@@ -57,16 +64,23 @@ export const TextArea = React.forwardRef<HTMLSpanElement, TextAreaProps>(functio
         onUpdate,
         onChange,
     } = props;
+
+    const {errorMessageProp, validationStateProp} = errorPropsMapper({
+        error,
+        errorMessage,
+        validationState,
+    });
+
     const [uncontrolledValue, setUncontrolledValue] = React.useState(defaultValue ?? '');
     const innerControlRef = React.useRef<HTMLTextAreaElement | HTMLInputElement>(null);
     const [hasVerticalScrollbar, setHasVerticalScrollbar] = React.useState(false);
-    const state = getInputControlState({error});
+    const state = getInputControlState(validationStateProp);
     const handleRef = useForkRef(props.controlRef, innerControlRef);
     const innerId = useUniqId();
 
     const isControlled = value !== undefined;
     const inputValue = isControlled ? value : uncontrolledValue;
-    const isErrorMsgVisible = typeof error === 'string';
+    const isErrorMsgVisible = Boolean(errorMessageProp);
```

Comment:
If validationState is not `invalid` we shouldn't show the error

Reply:
Looks like it's still an issue

Reply:
my bad
fixed

Reply:
Couldn't reproduce in the storybook, but according to code it's not fixed, if `errorMessage` is passed then `validationState` will be `invalid`.

How it should work:
First we map old prop to new ones
* `error = true` => `errorMessage = undefined` and `validationState = 'invalid'`
* `error = 'Some error'` => `errorMessage = 'Some error'` and `validationState = 'invalid'`

Then we check if both 1) `errorMessage` has a value and 2) `validationState = invalid` then we display an error, otherwise we won't.

Reply:
@saxumcordis  I forgot one statement, on [this](https://github.com/gravity-ui/uikit/pull/821/files#diff-e837ab9b77ac6a096b85a7932f340c6ff8419adb82f50c615ca6c52ef2e480a2R47) line you set `validationState = invalid` if `errorMessage` is not empty. New `errorMessage` prop should not set `validattionState`, only old `error` prop should. Also why empty strings in errors are considered as values? Empty string should be equal to undefined

Reply:
okay, got u

Reply:
done